include/avif/avif.h

```diff
@@ -48,6 +48,21 @@ extern "C" {
 #define AVIF_API
 #endif // defined(AVIF_DLL)
 
+#if defined(AVIF_ENABLE_NODISCARD) || (defined(__cplusplus) && __cplusplus >= 201700L) || \
+    (defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L)
+#define AVIF_NODISCARD [[nodiscard]]
+#else
+#if defined(__clang__) && defined(__has_attribute)
```

Comment:
It seems that newer versions of GCC also have the `__has_attribute` operator. See https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fattribute.html.

Reply:
Ah indeed, I added `defined(__clang__)` thinking it would help windows compilation.
Actually, the issue is that `cl.exe` does not do short-circuit evaluation for preprocessor with `defined(__clang__) && defined(__has_attribute) && __has_attribute(warn_unused_result)`.